Bottom: eeb1c29afd422cdcafd2a9d6753b03916c8a59d6
Top:    4a7506b5c449f88fd2cd017255387e7c69a1d3da
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-24 13:25:50 +0100

Refresh of hid-gtk-refactor-storage-seman

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 915656a..e38cea4 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -241,21 +241,21 @@ zoom_to (double new_zoom, int x, int y)
       gdouble xtmp, ytmp;
       gint x0, y0;
 
-      xtmp = (SIDE_X (gport->view_x) - gport->view_x0) /
+      xtmp = (SIDE_X (gport->pcb_x) - gport->view_x0) /
                 (gdouble) gport->view_width;
-      ytmp = (SIDE_Y (gport->view_y) - gport->view_y0) /
+      ytmp = (SIDE_Y (gport->pcb_y) - gport->view_y0) /
                 (gdouble) gport->view_height;
 
       gport->zoom = new_zoom;
       pixel_slop = new_zoom;
       ghid_port_ranges_scale(FALSE);
 
-      x0 = SIDE_X (gport->view_x) - xtmp * gport->view_width;
+      x0 = SIDE_X (gport->pcb_x) - xtmp * gport->view_width;
       if (x0 < 0)
         x0 = 0;
       gport->view_x0 = x0;
 
-      y0 = SIDE_Y (gport->view_y) - ytmp * gport->view_height;
+      y0 = SIDE_Y (gport->pcb_y) - ytmp * gport->view_height;
       if (y0 < 0)
         y0 = 0;
       gport->view_y0 = y0;
@@ -1518,12 +1518,12 @@ SwapSides (int argc, char **argv, int x, int y)
      location */
   if (do_flip_x)
     {
-	flipd = PCB->MaxWidth / 2 - SIDE_X (gport->view_x);
+	flipd = PCB->MaxWidth / 2 - SIDE_X (gport->pcb_x);
 	ghid_port_ranges_pan (2 * flipd, 0, TRUE);
     }
   if (do_flip_y)
     {
-	flipd = PCB->MaxHeight / 2 - SIDE_Y (gport->view_y);
+	flipd = PCB->MaxHeight / 2 - SIDE_Y (gport->pcb_y);
 	ghid_port_ranges_pan (0, 2 * flipd, TRUE);
     }
 
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 694a712..0a90931 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -166,21 +166,21 @@ ghid_port_ranges_zoom (gdouble zoom)
   if ((zoom > xtmp && zoom > ytmp) || zoom == 0.0)
     zoom = (xtmp > ytmp) ? xtmp : ytmp;
 
-  xtmp = (SIDE_X (gport->view_x) - gport->view_x0) /
+  xtmp = (SIDE_X (gport->pcb_x) - gport->view_x0) /
             (gdouble) gport->view_width;
-  ytmp = (SIDE_Y (gport->view_y) - gport->view_y0) /
+  ytmp = (SIDE_Y (gport->pcb_y) - gport->view_y0) /
             (gdouble) gport->view_height;
 
   gport->zoom = zoom;
   pixel_slop = zoom;
   ghid_port_ranges_scale(FALSE);
 
-  x0 = SIDE_X (gport->view_x) - xtmp * gport->view_width;
+  x0 = SIDE_X (gport->pcb_x) - xtmp * gport->view_width;
   if (x0 < 0)
     x0 = 0;
   gport->view_x0 = x0;
 
-  y0 = SIDE_Y (gport->view_y) - ytmp * gport->view_height;
+  y0 = SIDE_Y (gport->pcb_y) - ytmp * gport->view_height;
   if (y0 < 0)
     y0 = 0;
   gport->view_y0 = y0;
@@ -205,8 +205,8 @@ ghid_get_coords (const char *msg, int *x, int *y)
     ghid_get_user_xy (msg);
   if (ghid_port.has_entered)
     {
-      *x = gport->view_x;
-      *y = gport->view_y;
+      *x = gport->pcb_x;
+      *y = gport->pcb_y;
     }
 }
 
@@ -226,10 +226,10 @@ ghid_note_event_location (GdkEventButton * ev)
       event_x = ev->x;
       event_y = ev->y;
     }
-  gport->view_x = VIEW_X (event_x);
-  gport->view_y = VIEW_Y (event_y);
+  gport->pcb_x = VIEW_X (event_x);
+  gport->pcb_y = VIEW_Y (event_y);
 
-  moved = MoveCrosshairAbsolute (gport->view_x, gport->view_y);
+  moved = MoveCrosshairAbsolute (gport->pcb_x, gport->pcb_y);
   if (moved)
     {
       AdjustAttachedObjects ();
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 31ee636..efe6b3b 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -177,7 +177,8 @@ typedef struct
   gdouble zoom;			/* PCB units per screen pixel.  Larger */
   /* numbers mean zooming out. */
   gint view_x0,			/* Viewport in PCB coordinates */
-    view_y0, view_width, view_height, view_x, view_y;
+    view_y0, view_width, view_height;
+  Coord pcb_x, pcb_y;
 
   gint x_crosshair, y_crosshair;
 }
